pipeline/pipeline.go

```diff
@@ -261,37 +276,31 @@ func (pipeline *Pipeline) configuringValues(context *structContext) error {
 	case reflect.Slice, reflect.Map, reflect.Array:
 		valueGet := pipeline.chains.stageFunction.GetComplexType(context)
 		fmt.Println("Loglevel: Debug Message: [Pipeline]:  value get from parsing slice: ", valueGet)
-		if valueGet.CheckIsValue() {
-
-			// add check for setuping valueGet in valueIndirect
-			if valueIndirect.CanSet() {
-				fmt.Println("Loglevel: Debug Message: [Pipeline]: setupe value in struct")
-				valueIndirect.Set(valueGet.Value)
-			} else {
-				return errors.New("can not set " + valueIndirect.Kind().String() + " into struct field.")
-			}
-		} else {
-			return errors.New("Loglevel: Debug Message:  value get not implementedable value: ")
-		}
+		return pipeline.setupValue(context, &valueGet)
 	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
-		return errors.New("not supported types of unsigned integer")
+		return errors.New("not implemented types of unsigned integer")
 	case reflect.String, reflect.Float32, reflect.Float64, reflect.Bool, reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
 		valueGet := pipeline.chains.stageFunction.GetBaseType(context)
-		if valueGet.CheckIsValue() {
-			fmt.Println("Loglevel: Debug Message: [Pipeline]: value get from parsing "+valueGet.Value.Kind().String()+": ", valueGet)
-			if valueIndirect.CanSet() {
-				fmt.Println("[Pipeline]: setupe value")
-				valueIndirect.Set(valueGet.Value)
-			} else {
-				return errors.New("can not set " + valueIndirect.Kind().String() + " into struct field.")
-			}
-		} else {
-			return errors.New("value get not implementedable value: ")
-		}
+		return pipeline.setupValue(context, &valueGet)
 	default:
 		return errors.New("not supported type for hocon parsing")
 	}
-	return nil
+}
+
+func (pipeline *Pipeline) setupValue(context *structContext, value *infra.GoStructorValue) error {
+	valueIndirect := reflect.Indirect(context.Value)
+	if value.CheckIsValue() {
+		// add check for setuping valueGet in valueIndirect
+		if valueIndirect.CanSet() {
+			fmt.Println("Loglevel: Debug Message: [Pipeline]: setupe value in struct")
+			valueIndirect.Set(value.Value)
+			return nil
+		} else {
+			return errors.New("can not set " + value.Value.Kind().String() + " into struct field.")
+		}
+	} else {
+		return errors.New("Loglevel: Debug Message:  value get not implementedable value: ")
```

Comment:
I believe there must be error level not debug

Reply:
for all logs need refactore by resolve this issue: https://github.com/goreflect/gostructor/issues/12